### ISSUE TITLE
PPYOLOE loss_cl非常大

### ISSUE THREAD

**Issue by 875052544** [PaddlePaddle/PaddleDetection#6117]
### 问题确认 Search before asking

- [X] 我已经查询[历史issue](https://github.com/PaddlePaddle/PaddleDetection/issues)，没有报过同样bug。I have searched the [issues](https://github.com/PaddlePaddle/PaddleDetection/issues) and found no similar bug report.

### bug描述 Describe the Bug

[06/02 11:49:40] ppdet.utils.checkpoint INFO: Finish loading model weights: /root/.cache/paddle/weights/CSPResNetb_x_pretrained.pdparams
[06/02 11:49:42] ppdet.engine INFO: Epoch: [0] [  0/327] learning_rate: 0.000000 loss: 3.975326 loss_cls: 0.105371 loss_iou: 0.981339 loss_dfl: 2.833213 loss_l1: 7.127966 eta: 2 days, 5:15:06 batch_cost: 1.9542 data_cost: 0.0003 ips: 8.1875 images/s
[06/02 11:53:04] ppdet.engine INFO: Epoch: [0] [300/327] learning_rate: 0.003303 loss: 37.457855 loss_cls: 35.277142 loss_iou: 0.620637 loss_dfl: 1.260373 loss_l1: 0.674086 eta: 16:42:04 batch_cost: 0.6103 data_cost: 0.0001 ips: 26.2162 images/s
[06/02 11:53:27] ppdet.engine INFO: Epoch: [1] [  0/327] learning_rate: 0.003600 loss: 44.684166 loss_cls: 42.488293 loss_iou: 0.599753 loss_dfl: 1.204882 loss_l1: 0.606429 eta: 16:59:13 batch_cost: 0.6244 data_cost: 0.0123 ips: 25.6245 images/s
[06/02 11:56:43] ppdet.engine INFO: Epoch: [1] [300/327] learning_rate: 0.006903 loss: 57.273800 loss_cls: 55.637402 loss_iou: 0.469898 loss_dfl: 0.916840 loss_l1: 0.422670 eta: 16:30:10 batch_cost: 0.5921 data_cost: 0.0004 ips: 27.0246 images/s
[06/02 11:57:07] ppdet.engine INFO: Epoch: [2] [  0/327] learning_rate: 0.007200 loss: 57.396709 loss_cls: 55.745827 loss_iou: 0.467921 loss_dfl: 0.912606 loss_l1: 0.418009 eta: 16:41:23 batch_cost: 0.6147 data_cost: 0.0160 ips: 26.0311 images/s
[06/02 12:00:27] ppdet.engine INFO: Epoch: [2] [300/327] learning_rate: 0.010503 loss: 57.955795 loss_cls: 56.367802 loss_iou: 0.461489 loss_dfl: 0.912825 loss_l1: 0.432382 eta: 16:33:04 batch_cost: 0.6063 data_cost: 0.0001 ips: 26.3896 images/s
[06/02 12:00:47] ppdet.engine INFO: Epoch: [3] [  0/327] learning_rate: 0.010800 loss: 57.499207 loss_cls: 55.874950 loss_iou: 0.465653 loss_dfl: 0.918409 loss_l1: 0.439063 eta: 16:31:49 batch_cost: 0.6084 data_cost: 0.0068 ips: 26.2982 images/s
[06/02 12:04:10] ppdet.engine INFO: Epoch: [3] [300/327] learning_rate: 0.014103 loss: 53.663479 loss_cls: 51.858963 loss_iou: 0.507553 loss_dfl: 1.005520 loss_l1: 0.508582 eta: 16:30:19 batch_cost: 0.6169 data_cost: 0.0043 ips: 25.9350 images/s
[06/02 12:04:34] ppdet.engine INFO: Epoch: [4] [  0/327] learning_rate: 0.014400 loss: 53.663479 loss_cls: 51.858963 loss_iou: 0.509146 loss_dfl: 1.003465 loss_l1: 0.499418 eta: 16:34:53 batch_cost: 0.6175 data_cost: 0.0180 ips: 25.9
loss_cls: 已经达到50多了  昨晚我已经跑了80个批次 输出的map是0

### 复现环境 Environment

我用的版本是PPYOLOE的X版本 其中 S、L版本我都用过了 无这类情况发生
GPU:A5000 1个
PaddlePaddle  2.2.0
Python  3.8
Cuda  11.2
CPU  15核 AMD EPYC 7543 32-Core Processor

### 是否愿意提交PR Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by 875052544** [comment]
[06/02 13:10:46] ppdet.engine INFO: Epoch: [80] [  0/327] learning_rate: 0.016089 loss: 63.948833 loss_cls: 62.604023 loss_iou: 0.386131 loss_dfl: 0.758962 loss_l1: 0.285528 eta: 2 days, 10:36:25 batch_cost: 2.9328 data_cost: 1.0341 ips: 5.4555 images/s
[06/02 13:14:02] ppdet.engine INFO: Epoch: [80] [300/327] learning_rate: 0.016044 loss: 68.280579 loss_cls: 67.029465 loss_iou: 0.344272 loss_dfl: 0.753385 loss_l1: 0.277578 eta: 11:55:32 batch_cost: 0.5915 data_cost: 0.0002 ips: 27.0497 images/s
[06/02 13:14:26] ppdet.engine INFO: Epoch: [81] [  0/327] learning_rate: 0.016040 loss: 69.098663 loss_cls: 67.868271 loss_iou: 0.342881 loss_dfl: 0.753134 loss_l1: 0.277956 eta: 12:08:30 batch_cost: 0.6100 data_cost: 0.0121 ips: 26.2306 images/s
[06/02 13:17:46] ppdet.engine INFO: Epoch: [81] [300/327] learning_rate: 0.015994 loss: 8224.982422 loss_cls: 8223.742188 loss_iou: 0.340585 loss_dfl: 0.741088 loss_l1: 0.269461 eta: 12:01:27 batch_cost: 0.6033 data_cost: 0.0002 ips: 26.5196 images/s
[06/02 13:18:09] ppdet.engine INFO: Epoch: [82] [  0/327] learning_rate: 0.015990 loss: 8224.982422 loss_cls: 8223.742188 loss_iou: 0.341914 loss_dfl: 0.745785 loss_l1: 0.275981 eta: 12:06:26 batch_cost: 0.6155 data_cost: 0.0121 ips: 25.9967 images/s
[06/02 13:21:34] ppdet.engine INFO: Epoch: [82] [300/327] learning_rate: 0.015944 loss: 8606.712891 loss_cls: 8605.501953 loss_iou: 0.354925 loss_dfl: 0.768179 loss_l1: 0.288401 eta: 12:04:55 batch_cost: 0.6156 data_cost: 0.0002 ips: 25.9925 images/s
[06/02 13:22:01] ppdet.engine INFO: Epoch: [83] [  0/327] learning_rate: 0.015940 loss: 8513.798828 loss_cls: 8512.449219 loss_iou: 0.355533 loss_dfl: 0.765442 loss_l1: 0.286390 eta: 12:11:48 batch_cost: 0.6321 data_cost: 0.0143 ips: 25.3122 images/s
[06/02 13:25:25] ppdet.engine INFO: Epoch: [83] [300/327] learning_rate: 0.015893 loss: 8417.669922 loss_cls: 8416.386719 loss_iou: 0.348672 loss_dfl: 0.758343 loss_l1: 0.281732 eta: 12:08:18 batch_cost: 0.6173 data_cost: 0.0002 ips: 25.9178 images/s
[06/02 13:25:51] ppdet.engine INFO: Epoch: [84] [  0/327] learning_rate: 0.015889 loss: 8417.669922 loss_cls: 8416.386719 loss_iou: 0.347429 loss_dfl: 0.755357 loss_l1: 0.280692 eta: 12:12:24 batch_cost: 0.6368 data_cost: 0.0142 ips: 25.1266 images/s
继续训练 loss高达上千 一直升没降

**Comment by 875052544** [comment]
[06/02 13:38:07] ppdet.engine INFO: Epoch: [0] [  0/104] learning_rate: 0.000000 loss: 4.043756 loss_cls: 0.261828 loss_iou: 0.946129 loss_dfl: 2.833213 loss_l1: 6.674616 eta: 4 days, 11:23:10 batch_cost: 12.3907 data_cost: 9.6204 ips: 4.0353 images/s
[06/02 13:39:18] ppdet.engine INFO: Epoch: [0] [100/104] learning_rate: 0.003846 loss: 3.900480 loss_cls: 1.379240 loss_iou: 0.651883 loss_dfl: 1.289385 loss_l1: 1.515441 eta: 6:45:25 batch_cost: 0.6661 data_cost: 0.0158 ips: 75.0643 images/s
[06/02 13:39:36] ppdet.engine INFO: Epoch: [1] [  0/104] learning_rate: 0.004000 loss: 3.880305 loss_cls: 1.379240 loss_iou: 0.627971 loss_dfl: 1.263932 loss_l1: 1.303119 eta: 7:48:21 batch_cost: 0.7990 data_cost: 0.1525 ips: 62.5759 images/s
[06/02 13:40:49] ppdet.engine INFO: Epoch: [1] [100/104] learning_rate: 0.007846 loss: 2.903537 loss_cls: 1.185107 loss_iou: 0.491620 loss_dfl: 0.948516 loss_l1: 0.446540 eta: 6:54:51 batch_cost: 0.6974 data_cost: 0.0754 ips: 71.6937 images/s
[06/02 13:41:06] ppdet.engine INFO: Epoch: [2] [  0/104] learning_rate: 0.008000 loss: 2.898614 loss_cls: 1.179549 loss_iou: 0.489181 loss_dfl: 0.948252 loss_l1: 0.443910 eta: 7:21:03 batch_cost: 0.8139 data_cost: 0.1893 ips: 61.4349 images/s
[06/02 13:42:27] ppdet.engine INFO: Epoch: [2] [100/104] learning_rate: 0.011846 loss: 2.729434 loss_cls: 1.172084 loss_iou: 0.436166 loss_dfl: 0.897938 loss_l1: 0.381139 eta: 7:06:38 batch_cost: 0.7759 data_cost: 0.1098 ips: 64.4387 images/s
[06/02 13:42:41] ppdet.engine INFO: Epoch: [3] [  0/104] learning_rate: 0.012000 loss: 2.724624 loss_cls: 1.169245 loss_iou: 0.436166 loss_dfl: 0.883752 loss_l1: 0.378878 eta: 7:19:40 batch_cost: 0.8395 data_cost: 0.1864 ips: 59.5583 images/s

这是我训练ppyoloe-s的部分loss 很正常  为什么会出现这种情况？

**Comment by ghostxsl** [comment]
尝试一下不加--amp训练

**Comment by 875052544** [comment]
我看过另一篇帖子 尝试过没加amp 但是loss的结果还是飙升



------------------&nbsp;原始邮件&nbsp;------------------
发件人: "shangliang ***@***.***&gt;; 
发送时间: 2022年6月2日(星期四) 下午2:05
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [PaddlePaddle/PaddleDetection] PPYOLOE loss_cl非常大 (Issue #6117)





 
尝试一下不加--amp训练

**Comment by ghostxsl** [comment]
调小学习率试一下

**Comment by 875052544** [comment]
loss应该不和学习率有关吧？




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "PaddlePaddle/PaddleDetection"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月2日(星期四) 下午2:10
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [PaddlePaddle/PaddleDetection] PPYOLOE loss_cl非常大 (Issue #6117)





 
调小学习率试一下

**Comment by Nioolek** [comment]
这个问题我也出现过。原因出现在这里：
```
assigned_scores_sum = assigned_scores.sum()
loss_cls /= assigned_scores_sum
```
如果使用的卡数量特别少，且因为显存限制，只能把batchsize设置的很小，那么会偶发assigned_scores_sum特别小的情况，小到甚至接近于1。
这种情况就会导致loss_cls在这个batch特别的大，反向传播之后会导致下个iter中loss_cls更加飙升。
所以综合而言，解决措施：
1\增加batchsize
2\多卡训练
仅靠调小lr解决不了本质问题
